### ISSUE TITLE
Using POST body for search

### ISSUE THREAD

**Issue by nasht00** [elastic/elasticsearch-js#391]
Using the simplest query:

```
client.search({q:"hello world})
```

I see from the network traffic that the request is `POST` but using URL query parameters: `POST /_search?q=hello%20world HTTP/1.1`. The request has no body.

Following the simple example in the documentation (https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/quick-start.html):

```
client.search({
  index: 'twitter',
  type: 'tweets',
  body: {
    query: {
      match: {
        body: 'elasticsearch'
      }
    }
  }
})
```

I see from the network traffic that it is making an `OPTIONS` request instead:
`OPTIONS /twitter/tweets/_search HTTP/1.1`

I am using the jQuery build (v10.1.3 - 2016-01-26)

**Comment by spalger** [comment]
I imagine what you are seeing is the [preflight request sent by browsers for cross-domain requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests).

**Comment by nasht00** [comment]
Ok according to some research, it seems to be related to this issue:
https://github.com/elastic/elasticsearch/issues/17483

Fix to be release with 2.3.2 ...
